src/view/com/util/post-embeds/VideoEmbedInner/VideoWebControls.tsx

```diff
@@ -210,123 +255,333 @@ export function Controls({
           {error && (
             <Text style={{color: t.palette.white}}>
               <Trans>An error occurred</Trans>
             </Text>
           )}
-        </Animated.View>
+        </View>
       )}
     </div>
   )
 }
 
-const btnProps = {
-  variant: 'ghost',
-  shape: 'round',
-  size: 'medium',
-  style: a.p_2xs,
-  hoverStyle: {backgroundColor: 'rgba(255, 255, 255, 0.1)'},
-} as const
+function ControlButton({
+  active,
+  activeLabel,
+  inactiveLabel,
+  activeIcon: ActiveIcon,
+  inactiveIcon: InactiveIcon,
+  onPress,
+}: {
+  active: boolean
+  activeLabel: string
+  inactiveLabel: string
+  activeIcon: React.ComponentType<Pick<SvgProps, 'fill' | 'width'>>
+  inactiveIcon: React.ComponentType<Pick<SvgProps, 'fill' | 'width'>>
+  onPress: () => void
+}) {
+  const t = useTheme()
+  return (
+    <Button
+      label={active ? activeLabel : inactiveLabel}
+      onPress={onPress}
+      variant="ghost"
+      shape="round"
+      size="medium"
+      style={a.p_2xs}
+      hoverStyle={{backgroundColor: 'rgba(255, 255, 255, 0.1)'}}>
+      {active ? (
+        <ActiveIcon fill={t.palette.white} width={20} />
+      ) : (
+        <InactiveIcon fill={t.palette.white} width={20} />
+      )}
+    </Button>
+  )
+}
+
+function Scrubber({
+  duration,
+  currentTime,
+  onSeek,
+  onSeekEnd,
+  onSeekStart,
+  seekLeft,
+  seekRight,
+  togglePlayPause,
+  drawFocus,
+}: {
+  duration: number
+  currentTime: number
+  onSeek: (time: number) => void
+  onSeekEnd: () => void
+  onSeekStart: () => void
+  seekLeft: () => void
+  seekRight: () => void
+  togglePlayPause: () => void
+  drawFocus: () => void
+}) {
+  const {_} = useLingui()
+  const t = useTheme()
+  const {
+    state: scrubbing,
+    onIn: startScrubbing,
+    onOut: stopScrubbing,
+  } = useInteractionState()
+  const {
+    state: hovered,
+    onIn: onMouseEnter,
+    onOut: onMouseLeave,
+  } = useInteractionState()
+  const {state: focused, onIn: onFocus, onOut: onBlur} = useInteractionState()
+  const [seekPosition, setSeekPosition] = useState(0)
+  const isSeekingRef = useRef(false)
+  const ref = useRef<HTMLDivElement>(null)
+
+  useEffect(() => {
+    function seek(evt: MouseEvent | TouchEvent) {
+      if (!ref.current) return
+      const {left, width} = ref.current.getBoundingClientRect()
+      const x = 'touches' in evt ? evt.touches[0].clientX : evt.clientX
+      const percent = clamp((x - left) / width, 0, 1) * duration
+      onSeek(percent)
+      setSeekPosition(percent)
+    }
+
+    // Handle mousedown and touchstart on the specific element
+    const handleMouseDown = (evt: MouseEvent | TouchEvent) => {
+      if (!ref.current) return
+      const target = evt.target as Node | null
+      if (ref.current.contains(target)) {
+        evt.preventDefault()
+        isSeekingRef.current = true
+        seek(evt)
+        onSeekStart()
+        startScrubbing()
+      }
+    }
+
+    // Handle mouseup and touchend anywhere on the document
```

Comment:
This feels complicated. Can you use pointer events + setPointerCapture?

Reply:
didn't know that was a thing! thanks, will do